configure.ac

```diff
@@ -762,35 +762,18 @@ fi
 dnl Check valgrind support.
 PHP_ARG_WITH([valgrind],
   [whether to enable valgrind support],
-  [AS_HELP_STRING([--with-valgrind=DIR],
+  [AS_HELP_STRING([--with-valgrind],
     [Enable valgrind support])],
   [yes],
   [no])
 
 if test "$PHP_VALGRIND" != "no"; then
+  PKG_CHECK_MODULES([VALGRIND], [valgrind], [have_valgrind="yes"], [have_valgrind="no"])
 
-  AC_MSG_CHECKING([for valgrind header])
-
-  if test "$PHP_VALGRIND" = "yes"; then
-    SEARCH_PATH="/usr/local /usr"
-  else
-    SEARCH_PATH="$PHP_VALGRIND"
-  fi
-
-  SEARCH_FOR="/include/valgrind/valgrind.h"
-  for i in $SEARCH_PATH ; do
-    if test -r $i/$SEARCH_FOR; then
-      VALGRIND_DIR=$i
-    fi
-  done
-
-  if test -z "$VALGRIND_DIR"; then
-    AC_MSG_RESULT([not found])
-  else
-    AC_MSG_RESULT(found in $VALGRIND_DIR)
+  if test "x$have_valgrind" = "xyes"; then
+    PHP_EVAL_INCLINE($VALGRIND_CFLAGS)
```

Comment:
It would be nice to generate an error if `--with-valgrind` has been explicitly given (not auto-enabled) and valgrind couldn't be found.

Reply:
Do you still want valgrind support enabled by default?

Reply:
Yeah, I think so. Basically:

 * Default: Valgrind used if available, no error if not available.
 * `--with-valgrind`: Valgrind must be available, error otherwise.
 * `--without-valgrind`: Valgrind not used even if available.